docs/develop/excel-async-model.md

```diff
@@ -0,0 +1,180 @@
+
+To use the async APIs, you need to add additional syntax to parts of your script. This tells the editor which code blocks are calling the async APIs. You need to add `async` to your `main` function and surround async commands in a function passed to `Excel.run`.
+
+## `main` function
+
+To use the async APIs, your script's `main` function needs to be an `async` function, as shown in the following code:
+
+```TypeScript
+async function main(workbook: ExcelScript.Workbook) {
+    // Your Office Script
+}
+```
+
+## `Excel.run` blocks for async code
+
+`Excel.run` is a function that runs async Office Scripts code. The following script shows how we recommend using `Excel.run`. Note the following:
+
+- We use `await` before `Excel.run`. This ensures the async block of code completes before proceeding. Otherwise, we might have conflicts with our own script.
+- The function passed to `Excel.run` is also async. This allows it to run without explicitly returning a [Promise](https://developer.mozilla.org/docs/web/javascript/reference/global_objects/promise). In the `Excel.run` block, you can set any variables that were declared earlier in the script. You can also return a value from the function from `await`ed `run` block. Be aware that the Excel objects are different in API sets are not the same and cannot be translated to and from async in your script.
+- Additional `Excel.run` calls could be made later in the script. You could also pull them into separate functions called by `main`.
+
+```TypeScript
+async function main(workbook: ExcelScript.Workbook) {
+
+    // Standard scripting code...
+
+    // A standard async API block.
+    await Excel.run(async (context: Excel.RequestContext) => {
+        // Async API code...
+    });
+
+    // More standard scripting code...
+
+    // An async API block with a returned value.
+    let x = await Excel.run(async (context: Excel.RequestContext) => {
+        // Async API code...
+        return myValue;
+    });
+}
+```
+
+## Context
+
+The `main` function accepts an `Excel.RequestContext` parameter, named `context`. Think of `context` as the bridge between your script and the workbook. Your script accesses the workbook with the `context` object and uses that `context` to send data back and forth.
+
+The `context` object is necessary because the script and Excel are running in different processes and locations. The script will need to make changes to or query data from the workbook in the cloud. The `context` object manages those transactions.
+
+By default, Office Scripts handle the interactions between your script and the workbook automatically. While the process is optimized, the standard Office Scripts APIs may synchronize the workbook with your script more than necessary, such as during looped read operations. You may be able to manage these workbook-script data transactions more efficiently with the async APIs, since you know when it is necessary to update data.
```

Comment:
```suggestion
By default, Office Scripts handle the interactions between your script and the workbook automatically. If your code doesn't have performance-related issues with synchronization, such as a looped read operations, then we recommend using [the synchronous API instead.](LINK)
```
If ENG can't think of why then I don't think we should write about that either, maybe just keep steering people away